server/routes/messagesRoutes.js

```diff
@@ -0,0 +1,131 @@
+const express = require('express');
+const jwt = require('jsonwebtoken');
+const mongoose = require('mongoose');
+const router = express.Router();
+const Message = require('../models/messageModel');
+const Conversation = require('../models/conversationModel');
+const verifyToken = require("../utils/verifyToken")
+
+let jwtUser = null;
+
+// Token verfication middleware
+router.use(function (req, res, next) {
+    try {
+        jwtUser = verifyToken(req.headers.authorization)
+        next();
+    } catch (err) {
+        console.log(err);
+        res.setHeader('Content-Type', 'application/json');
+        res.end(JSON.stringify({ message: 'Unauthorized' }));
+        res.sendStatus(401);
+    }
+});
+
+
+// Get messages from conversation
+// based on to & from
+router.get('/conversations/query', (req, res) => {
+    let user1 = mongoose.Types.ObjectId(jwtUser.id);
+    let user2 = mongoose.Types.ObjectId(req.query.userId);
+    Message.aggregate([
```

Comment:
You really should be getting messages from a conversation, using the conversation id, it'll be more reliable and more efficient to do so

Reply:
I will add this to my TODO's as an optimization for the future fixes.